File: prime_factor/primeFactor.js
function primeFactor (n) {
	/*
	Finding all the prime factors of a given number
	:param n: Number whose prime factors are going to be found
	:returns: Array with prime numbers
	 */
	let primes = [];
	for (let i = 2; i <= Math.sqrt(n); i++) {
		if (n % i === 0) {
			primes.push(i);
			while (n % i === 0) {
				n = n / i;
			}
		}
	}
	if (n !== 1) {
		primes.push(n);
	}
	return primes;
}

function main () {
	let n = 582;
	console.log('Prime factors are:');
	let primes = primeFactor(n);
	for (let i = 0; i < primes.length; i++) {
		console.log(primes[i]);
	}
}

main();

File: prime_factor/PrimeFactor.java
import java.util.ArrayList;

public class PrimeFactor {

    public static ArrayList<Integer> primeFactor(int n) {
        ArrayList<Integer> primeNo = new ArrayList<Integer>();
        for (int i = 2; i <= n; i++) {
            if (n % i == 0) {
                primeNo.add(i);
                while (n % i == 0) {
                    n = n / i;
                }
            }
        }
        return primeNo;
    }

    public static void main(String[] args) {
        int n = 8;
        System.out.println("Prime Factors are:");
        for (Integer i : primeFactor(n)) {
            System.out.println(i);
        }
    }
}
